src/language_server/document.rs

```diff
@@ -0,0 +1,213 @@
+use tower_lsp::lsp_types::TextDocumentContentChangeEvent;
+
+#[derive(Debug)]
+pub struct Document {
```

Comment:
Sure.  A Document represents a source code file in LSP.  However, a document itself is "representationless" in LSP, instead being only the file contents as a string.  We need to track version information for text synchronisation, hence the separate structure.

In a multilingual language server, it would need to be extended with what the document's language is (e.g. gleam, toml).